SignalServiceKit/src/Util/OWSAsserts.h

```diff
@@ -19,55 +19,56 @@ NS_ASSUME_NONNULL_BEGIN
 // OWSCAssert() and OWSCFail() should be used in free functions.
 
 #define OWSAssert(X)                                                                                                   \
-    if (!(X)) {                                                                                                        \
-        DDLogError(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                        \
-        [DDLog flushLog];                                                                                              \
-        NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                               \
-    }
+    do {                                                                                                               \
+        if (!(X)) {                                                                                                    \
+            DDLogError(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                    \
+            [DDLog flushLog];                                                                                          \
+            NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                           \
+        }                                                                                                              \
+    } while (NO)
```

Comment:
I like this construct which enforces the caller add a semicolon to the macro invocation.